src/ObjectUtils/LinuxMap.cpp

```diff
@@ -39,34 +41,42 @@ ErrorMessageOr<ModuleInfo> CreateModule(const std::filesystem::path& module_path
   if (!std::filesystem::exists(module_path)) {
     return ErrorMessage(absl::StrFormat("The module file \"%s\" does not exist", module_path));
   }
+
   std::error_code error;
   uint64_t file_size = std::filesystem::file_size(module_path, error);
   if (error) {
     return ErrorMessage(
         absl::StrFormat("Unable to get size of \"%s\": %s", module_path, error.message()));
   }
 
-  ErrorMessageOr<std::unique_ptr<ElfFile>> elf_file_or_error = CreateElfFile(module_path);
-  if (elf_file_or_error.has_error()) {
-    return ErrorMessage(
-        absl::StrFormat("Unable to load module: %s", elf_file_or_error.error().message()));
-  }
-
-  ErrorMessageOr<uint64_t> load_bias_or_error = elf_file_or_error.value()->GetLoadBias();
-  // Every loadable module contains a load bias.
-  if (load_bias_or_error.has_error()) {
-    return load_bias_or_error.error();
+  auto object_file_or_error = CreateObjectFile(module_path);
+  if (object_file_or_error.has_error()) {
+    return ErrorMessage(absl::StrFormat("Unable to create module from object file: %s",
+                                        object_file_or_error.error().message()));
   }
 
   ModuleInfo module_info;
-  module_info.set_name(elf_file_or_error.value()->GetName());
-  module_info.set_soname(elf_file_or_error.value()->GetSoname());
   module_info.set_file_path(module_path);
   module_info.set_file_size(file_size);
   module_info.set_address_start(start_address);
   module_info.set_address_end(end_address);
-  module_info.set_build_id(elf_file_or_error.value()->GetBuildId());
-  module_info.set_load_bias(load_bias_or_error.value());
+  module_info.set_name(object_file_or_error.value()->GetName());
+
+  if (object_file_or_error.value()->IsElf()) {
+    auto* elf_file = dynamic_cast<ElfFile*>((object_file_or_error.value().get()));
+    module_info.set_build_id(elf_file->GetBuildId());
```

Comment:
this is kind of important to have for coff file too, in case this is available. We want to live in the world where not having build_id is a rare exception.

Reply:
Yes, totally agree. Will add this as one of the next steps.